### PR TITLE
Add driver logs to Jobs page for submission jobs

### LINKED ISSUE #34187: [Core|Dashboard] Support custom tags for jobs.
### Description

It will be very helpful to allow users to add custom tags to ray jobs at the time of submission, and see those tags in job metadata on the dashboard. If there is a way of exporting this data or storing it in an external durable storage, this metadata will become a great resource for fault, latency, resource analysis and reporting.

The tags can be used for grouping, filtering and reporting purposes. This can also be used for enriching ML experiment tracking and model lineage, as well as cost attribution.

### Use case

We run multi-tenant ray clusters to balance between the number of idle ray clusters latency of creating a new cluster on-demand. This means there are multiple users from various teams/projects submitting jobs to a given ray cluster. In this situation, the only definite information the users know is the ray job_id and they have to track their jobs in a separate database to track their work for the team or project.

Following challenges are hard to address without any metadata on the jobs.
1. Group jobs by users, teams or projects. Report an aggregated view jobs.
2. Attribute jobs to a user, team or model. 
3. Attribute cost to a user, team or model. 
4. Estimate resource requirements or plan budgets at team or project level.
5. Identify common patterns of job failures or delays, or offenders of compute resources across jobs.
6. Analyze the job data at user, team or project level to get insights and make recommendations.
7. Enrich the data and model lineage graphs with ray jobs and associated metadata for better experiment tracking and reproducibility.

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
- Add driver logs to Jobs page for submission jobs
- Adds a refresh button to the log viewer to reload the logs.
- Refactors the log viewer from the logs page into its own component
- Updates the look and feel of the jobs page to match the new IA style.
- Adds User-provided metadata to the job detail page. (fixes #34187 )
- Updates the table icon 
- Change "Tasks" to "Tasks/actor overview"
- Adds Node Count Card next to ray status cards

Job detail page
Screenshot 2023-04-17 at 11 40 53 PM

Other job pages (new look and feel)
Screenshot 2023-04-17 at 11 19 35 PM
Screenshot 2023-04-17 at 11 19 38 PM

Jobs user-provided metadata
Screenshot 2023-04-18 at 12 00 45 AM
Screenshot 2023-04-18 at 12 00 52 AM

Autoscaler section
Screenshot 2023-04-18 at 12 29 16 AM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — COMMENTED** [review]
Is there any way to add a refresh button? Until we make it print last N lines & that's probably the best way to use this effectively.

**Comment by scottsun94** [comment]
1. The position of the "Expand" button is a bit weird. Can we put it always next to the text?
Screen Shot 2023-04-17 at 11 27 23 PM
Screen Shot 2023-04-17 at 11 28 37 PM
2. Shall we say "User-provided metadata" since the metadata can be arbitrary data. "User metadata" feels like that they are related to the user
3. It seems that the `metadata` is only available in jobs SDK. Is it something we should support in CLI?

File: dashboard/client/src/pages/job/JobDriverLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+) => {
+  const { ipLogMap } = useContext(GlobalContext);
+  const { driver_agent_http_address, driver_info, submission_id } = job;
+  const host = (() => {
+    if (driver_agent_http_address) {
+      return `${driver_agent_http_address}/logs/`;
+    } else if (driver_info && ipLogMap[driver_info.node_ip_address]) {
+      return `${ipLogMap[driver_info.node_ip_address]}/`;
+    }
+  })();
```
[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/client/src/pages/job/JobDriverLogs.tsx:23]: No in that case, we do not show any logs so the default behavior of `return undefined` is what we want.

File: dashboard/client/src/pages/job/JobDriverLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    UnifiedJob,
+    "driver_agent_http_address" | "driver_info" | "submission_id"
+  >,
+) => {
+  const { ipLogMap } = useContext(GlobalContext);
+  const { driver_agent_http_address, driver_info, submission_id } = job;
+  const host = (() => {
+    if (driver_agent_http_address) {
+      return `${driver_agent_http_address}/logs/`;
+    } else if (driver_info && ipLogMap[driver_info.node_ip_address]) {
```
[the remark this answers is not in the document]
Comment (alanwguo) [reply dashboard/client/src/pages/job/JobDriverLogs.tsx:20]: no for `driver_info.node_ip_address`, yes for the second thing. That's why we check for the value from the dictionary.

**Comment by alanwguo** [comment]
@scottsun94 , want to take another look? I think I added the autoscaler row to the job detail page since you last reviewed

**Comment by scottsun94** [comment]
I cannot seem to see the "User-provided metadata" field. Is that only available when it's not empty? Shall we just always show it? It might raise the awareness of this feature.
Screen Shot 2023-04-18 at 10 30 41 PM

**Comment by alanwguo** [comment]
It's hidden for driver jobs because it's not possible to set metadata.

For submission jobs, it shows up as a "-" if empty.

**Comment by scottsun94** [comment]
oh. I see. This is the same for driver logs.
Just throwing a thought: shall we actually expose both of them and add message bars/tooltips whichever is appropriate to educate users why it's not available and the difference between driver jobs and submission jobs? Otherwise, the control logic here is just mystical to them because they may assume all ray jobs should perform consistently.

**Comment by scottsun94** [comment]
Everything LGTM. Thanks!

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**